src/Build/BackEnd/BuildManager/BuildParameters.cs

```diff
@@ -801,6 +804,30 @@ public string OutputResultsCacheFile
             set => _outputResultsCacheFile = value;
         }
 
+#if FEATURE_REPORTFILEACCESSES
+        /// <summary>
+        /// Gets or sets a value indicating whether file accesses should be reported to any configured project cache plugins.
+        /// </summary>
+        public bool ReportFileAccesses
+        {
+            get => _reportFileAccesses;
+            set
+            {
+                _reportFileAccesses = value;
+
+                // TODO dfederm: What if either of these are set after ReportFileAccesses is? Do we need to move this elsewhere?
```

Comment:
I'm taking a slightly different route. I'm just moving this to `BuildManager.BeginBuild`. It just forces these to the correct values when `ReportFileAccesses` is enabled.

That's... maybe going against the users wishes, but also just doesn't allow them to do the wrong thing. Without making everything more complex like tracking whether these params were explicitly set or not, I think this might be an OK solution.